core/trino-spi/src/main/java/io/trino/spi/exchange/Exchange.java

```diff
@@ -55,22 +54,32 @@
      * @param sinkHandle - handle returned by <code>addSink</code>
      * @param taskAttemptId - attempt id
      * @return ExchangeSinkInstanceHandle to be sent to a worker that is needed to create an {@link ExchangeSink} instance using
-     * {@link ExchangeManager#createSink(ExchangeSinkInstanceHandle)}
+     * {@link ExchangeManager#createSink(ExchangeSinkInstanceHandle, Runnable)}
      */
     ExchangeSinkInstanceHandle instantiateSink(ExchangeSinkHandle sinkHandle, int taskAttemptId);
 
+    /**
+     * Update {@link ExchangeSinkInstanceHandle}. Update is requested by {@link ExchangeSink}.
+     * The updated {@link ExchangeSinkInstanceHandle} is expected to be set by {@link ExchangeSink#updateHandle(ExchangeSinkInstanceHandle)}.
+     *
+     * @param sinkHandle - handle returned by <code>addSink</code>
+     * @param taskAttemptId - attempt id
+     * @return updated handle
+     */
+    ExchangeSinkInstanceHandle updateSinkInstanceHandle(ExchangeSinkHandle sinkHandle, int taskAttemptId);
```

Comment:
As this is called in context of specific task, which was already assigned a `ExchangeSinkInstanceHandle` I think it is more natural for this method to just take `ExchangeSinkInstanceHandle` and no attemptId.
Can you elaborate why you chose this appraoch? We can chat offline.

Reply:
Discussed offline. This was changed to avoid extra uncertainty what `ExchangeSinkInstanceHandle` must be passed (an original one, updated one or either?)